parlai/mturk/tasks/acute_eval/frontend/components/custom.jsx

```diff
@@ -0,0 +1,514 @@
+  }
+
+  handleInputChange(event) {
+    console.log(event);
+    let target = event.target;
+    let value = target.value;
+    let name = target.name;
+
+    this.setState(
+      {[name]: value},
+      this.checkValidData
+    );
+  }
+
+  onSliderChange(val) {
+    console.log(val);
+    this.setState(
+      {['speakerChoice']: val},
+      this.checkValidData
+    );
+  }
+
+  handleEnterKey(event) {
+    event.preventDefault();
+    if (this.props.task_done) {
+      this.props.allDoneCallback();
+    } else if (this.props.subtask_done && this.props.show_next_task_button) {
+      this.props.nextButtonCallback();
+    }
+  }
+
+  render() {
+    if (this.props.current_subtask_index != null && this.props.current_subtask_index !== this.state.subtaskIndexSeen) {
+      this.setState(
+        {
+          subtaskIndexSeen: this.props.current_subtask_index,
+          textReason: '',
+          speakerChoice: '',
+        },
+      );
+    }
+    if (this.props.task_data === undefined ||
+        this.props.task_data.task_specs === undefined){
+      return (
+        <div></div>
+      );
+    }
+    let s1_choice = this.props.task_data.task_specs.s1_choice.split('<Speaker 1>');
+    let s2_choice = this.props.task_data.task_specs.s2_choice.split('<Speaker 2>');
+    let form_question = this.props.task_data.task_specs.question;
+    let text_question = "Please provide a brief justification for your choice (a few words or a sentence)";
+    let text_reason = (
+      <div>
+        <ControlLabel>{text_question}</ControlLabel>
+        <FormControl
+          type="text"
+          id="id_text_input"
+          name="textReason"
+          style={{width: '80%', height: '100%', float: 'left', 'fontSize': '16px'}}
+          value={this.state.textReason}
+          placeholder="Please enter here..."
+          onChange={this.handleInputChange}
+          />
+        </div>
+    );
+    let speaker1_div = <div style={speaker1_style}>Speaker 1</div>
+    let speaker2_div = <div style={speaker2_style}>Speaker 2</div>
+    let choice1 = <div>{s1_choice[0]}{speaker1_div}{s1_choice[1]}</div>;
+    let choice2 = <div>{s2_choice[0]}{speaker2_div}{s2_choice[1]}</div>;
+    return (
+      <div
+        id="response-type-text-input"
+        className="response-type-module"
+        style={{'paddingTop': '15px',
+                'float': 'left',
+                'width': '100%',
+                'backgroundColor': '#eeeeee'}}>
+            <Form
+              horizontal
+              style={{backgroundColor: '#eeeeee', paddingBottom: '10px'}}
+              onSubmit={this.handleEnterKey}
+              >
+              <div
+                className="container"
+                style={{'width': 'auto',}}>
+                <ControlLabel> {form_question} </ControlLabel>
+                <FormGroup>
+                    <Col sm={6}>
+                        <Radio
+                          name="speakerChoice"
+                          value="1"
```

Comment:
it'd be better if these are the model names, so what we dump to disk later is more easily analyzed.